app/models/user.rb

```diff
@@ -54,22 +54,53 @@ def administrated_projects
     scope
   end
 
-  def self.to_csv
-    @users = User.order(:id)
+  def self.to_csv(
+    inherited: false, deleted: false, project_id: nil, user_id: nil,
+    datetime: (Time.now.strftime "%Y%m%d_%H%M")
+  )
+    inherited = true if project_id || user_id
+    users = if user_id.nil?
+      deleted ? User.unscoped.order(:id) : User.order(:id)
```

Comment:
`unscoped` can do surprising things sometimes, I think it reflects your intention better to do `User.with_deleted { User.order(:id) }`